### PR TITLE
Ashwini

### PR DESCRIPTION
**Pull request by AshwiniShivaprasad**

### CODE REVIEW

File: src/test/java/com/vmware/gto/vcg/dataValidationTests/VSANHDDDataValidationTests.java [hunk 1]
```diff
@@ -1,227 +1,227 @@
...
+			cf.waitForVisible(homepage.btn_backToSearchResults);
+		}
+
+		try {
+
+			// Validate UI Model details and DB details
+			WebElement ui_getHDDModel = driver.findElement(By.className("details_table_tab1"));
+			String ui_getIOModelText = ui_getHDDModel.getText();
+
+			q = vsanhddQueries.getHDDModelDetails(pid);
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/dataValidationTests/VSANHDDDataValidationTests.java]: This change should be in separate review

File: src/test/java/com/vmware/gto/vcg/dbutil/VSANHDDQueries.java [hunk 2]
```diff
@@ -1,25 +1,230 @@
...
...HEN 'N/A' WHEN cd.FirmwareVersion = '' THEN 'N/A' ELSE cd.FirmwareVersion END AS FirmwareVersion FROM Component_Release cr INNER JOIN Releases r ON r.Release_Id = cr.Release_Id INNER JOIN HDD hdd ON hdd.HDD_Id = cr.Component_Id INNER JOIN Components c ON c.Component_Id = hdd.HDD_Id INNER JOIN Cert_Details cd ON cd.Component_Release_Id = cr.Component_Release_Id INNER JOIN Device_Type dt ON dt.DeviceType_Id = hdd.DeviceType_Id INNER JOIN Cert_Feature_Mapping cftm ON cftm.CertDetail_Id = cd.CertDetail_Id LEFT JOIN Features_Category fc ON fc.Category_Id = cftm.FeatureCategory_Id LEFT JOIN ProductFeatures pf ON pf.Feature_Id = cftm.Feature_Id LEFT JOIN Cert_KB_Mapping ckbm ON ckbm.CertDetail_Id = cd.CertDetail_Id LEFT JOIN TestConfiguration tc ON cd.Configuration_Id = tc.Configuration_Id LEFT JOIN OS_Cert_Patch_Mapping oscpm ON oscpm.CertDetail_Id = cd.CertDetail_Id LEFT JOIN Cert_Footnote_Mapping cfm ON cfm.CertDetail_Id = cd.CertDetail_Id LEFT JOIN Footnotes f ON (f.Footnote_Id = cfm.Footnote_Id AND f.ApprovalStatus = 'Approved') WHERE cr.Component_Id = "+pid+" AND r.Release_Id != 243 GROUP BY cd.CertDetail_Id ORDER BY r.SortOrder DESC , cd.HCLPostedDate DESC , tc.Configuration ASC;";
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/dbutil/VSANHDDQueries.java]: If we can remove those column which we are not using in our test case for validation

File: src/test/java/com/vmware/gto/vcg/APItests/HDDCertDetailsValidationTests.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	public void setup() {
+		vcgCertSearchAPIBaseURL = getConfigParameter("vcgCertSearchAPIBaseURL");
+		hddQueries = new VSANHDDQueries("test", "vsan_hcl20");
+		Assert.assertNotNull(dbUtil.getCon());
+	}
+
+	@Test(dataProvider = "hddProducts")
+	public void programProductIDAPI(int CertId) throws SQLException
+	{
+		CertId=4267170;
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/HDDCertDetailsValidationTests.java]: need to remove this value

File: src/test/java/com/vmware/gto/vcg/dataValidationTests/VSANHDDDataValidationTests.java [hunk 4]
```diff
@@ -1,227 +1,227 @@
...
-				driver.close();
-				driver.switchTo().window(parentWindow);
-			}
-
-		}
-	}
-
-}
+package com.vmware.gto.vcg.dataValidationTests;
+
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/dataValidationTests/VSANHDDDataValidationTests.java]: this file changes should be in separate review

File: src/test/java/com/vmware/gto/vcg/dbutil/VSANIOQueries.java [hunk 5]
```diff
@@ -1,23 +1,186 @@
...
...pe_Group_Id ASC) AS CHAR) AS IODevice_Type_Group_Id, CAST(GROUP_CONCAT(DISTINCT itg.Group_Name ORDER BY itg.Group_Name ASC) AS CHAR) AS Group_Name, GROUP_CONCAT(DISTINCT f.Footnote SEPARATOR '</li><li>') AS Footnotes FROM Components c INNER JOIN IODevices io ON io.IODevice_Id = c.Component_Id INNER JOIN Partners p ON p.Partner_Id = c.Partner_Id INNER JOIN Device_Type dt on dt.DeviceType_Id = io.DeviceType_Id INNER JOIN Component_Release cr ON cr.Component_Id = io.IODevice_Id INNER JOIN Cert_Details cd ON cd.Component_Release_Id = cr.Component_Release_Id LEFT JOIN Comp_Footnote_Mapping cfm ON cfm.Component_Id = c.Component_Id LEFT JOIN Footnotes f ON (f.Footnote_Id = cfm.Footnote_Id AND f.ApprovalStatus = 'Approved') LEFT JOIN IODevice_Type_Group_Mapping itgm ON itgm.IODevice_Id = io.IODevice_Id LEFT JOIN IODevice_Type_Group itg ON itgm.IODevice_Type_Group_Id = itg.IODevice_Type_Group_Id LEFT JOIN Cert_Feature_Mapping cfm2 ON cfm2.CertDetail_Id = cd.CertDetail_Id LEFT JOIN Features_Category fc ON cfm2.FeatureCategory_Id = fc.Category_Id LEFT JOIN ProductFeatures pf ON pf.Feature_Id = cfm2.Feature_Id WHERE c.Component_Id ="+pid+"  GROUP BY c.Component_Id ORDER BY p.PartnerName ASC;";
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/dbutil/VSANIOQueries.java]: We could remove those columns from query which we are not using for validation

File: src/test/java/com/vmware/gto/vcg/dbutil/VSANHDDQueries.java (hunk 2 above)
Comment (AshwiniShivaprasad) [reply src/test/java/com/vmware/gto/vcg/dbutil/VSANHDDQueries.java]: These query are used by data validation test cases also.

File: src/test/java/com/vmware/gto/vcg/dbutil/VSANIOQueries.java (hunk 5 above)
Comment (AshwiniShivaprasad) [reply src/test/java/com/vmware/gto/vcg/dbutil/VSANIOQueries.java]: These query are used by data validation test cases also.

File: src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java [hunk 6]
```diff
@@ -95,51 +95,56 @@ public void programProductIDAPI(int ProductId) throws SQLException {
...
-		  softAssert.assertEquals(rs.getString("formFactor"), content.get("formFactor"),"formfactor");
-		  softAssert.assertEquals(rs.getString("interFaceSpeedId"), content.get("interFaceSpeedId").toString(),"InterfaceSpeedId");
-		  softAssert.assertEquals(rs.getString("interFaceSpeed"), content.get("interFaceSpeed").toString(),"InterfaceSpeed");
 		  softAssert.assertEquals(rs.getString("uniqueId"), content.get("uniqueId"),"uniqueId");
+		  softAssert.assertEquals(rs.getString("deviceTypeId"), content.get("deviceTypeId").toString(),"DeviceaTypeId");
+		  softAssert.assertEquals(rs.getString("ssdPartNumber"), content.get("ssdPartNumber"),"ssdPartNumber");
 		  softAssert.assertEquals(rs.getString("capacity"), content.get("capacity").toString(),"capacity");
+		  //softAssert.assertEquals(rs.getString("ssdTypes"), content.get("ssdTypes"));            //need to parse
+		  softAssert.assertEquals(rs.getString("ssdSeries"), content.get("ssdSeries"),"ssdSeries");
+		  softAssert.assertEquals(rs.getString("deviceTypeId"), content.get("deviceTypeId").toString(),"deviceTypeId");
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java]: this is already getting asserted @ line 105

File: src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java [hunk 7]
```diff
@@ -95,51 +95,56 @@ public void programProductIDAPI(int ProductId) throws SQLException {
...
 		  softAssert.assertEquals(rs.getString("enduranceClassId"), content.get("enduranceClassId").toString(),"enduranceClassId");
 		  softAssert.assertEquals(rs.getString("endurance"), content.get("endurance").toString(),"Endurance");
 		  softAssert.assertEquals(rs.getString("flashTechId"), content.get("flashTechId").toString(),"flashtechId");
 		  softAssert.assertEquals(rs.getString("technology"), content.get("technology"),"technology");
-		  softAssert.assertEquals(rs.getString("tbwSSDEnduranceClassId"), content.get("tbwSSDEnduranceClassId").toString(),"EnduranceClassId");
-		  softAssert.assertEquals(rs.getString("enduranceTBW"), content.get("enduranceTBW").toString(),"EnduranceClass");
 		  softAssert.assertEquals(rs.getString("ioDeviceId"), content.get("ioDeviceId").toString(),"ioDeviceId");
-		  softAssert.assertEquals(rs.getString("ioDevice"), content.get("ioDevice"),"ioDevice");
-		  softAssert.assertEquals(rs.getString("ssdSeries"), content.get("ssdSeries"),"ssdSeries");
+		  //softAssert.assertEquals(rs.getString("ioDeviceModel"), content.get("ioDeviceModel"),"ioDeviceModel");
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java]: why is it commented?

File: src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java [hunk 8]
```diff
@@ -95,51 +95,56 @@ public void programProductIDAPI(int ProductId) throws SQLException {
 	      log.info("From db " + columnName + " : " + columnValues.get(columnName));
 	      columnCount--;
 	    }
-		  softAssert.assertEquals(rs.getString("program"), content.get("program"),"Program ");
+
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java]: We have to add validation for "releaseVersionOld" value as well

File: config/config.properties [hunk 9]
```diff
@@ -1,22 +1,22 @@
...
+ChromeDriverPath=Resources\\Selenium\\drivers\\chromedriver.exe
+FireFoxPath=Resources\\Selenium\\drivers\\geckodriver.exe
+safaraiPath=Resources\\Selenium\\drivers\\safari.exe
+IEDriverPath=Resources\\Selenium\\drivers\\IEDriverServer.exe
+microSleepTime = 1000
+minSleepTime=3000
+maxSleepTime=5000
+waitTime=2
+isUITest=true
+vcgProductsAPIBaseURL=https://vcgapigateway-stage.apps.itcna.vmware.com/productsapi/v1
```
Comment (ashirvadkum) [inline config/config.properties]: we could make the link value format consistent by add '/' at the end or remove from all 3 values

File: src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java [hunk 10]
```diff
@@ -17,41 +17,40 @@
...
+		ssdQueries = new VSANSSDQueries("test", "vsan_hcl20");
 		Assert.assertNotNull(dbUtil.getCon());
 	}
 
-	@Test(dataProvider = "hddProducts")
+	@Test(dataProvider = "ssdProducts")
 	public void programProductIDAPI(int CertId) throws SQLException
 	{
-		CertId=4267170;
+		CertId=4281988;
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java]: hardcoded value should be removed

File: src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java [hunk 11]
```diff
@@ -61,49 +60,101 @@ public void programProductIDAPI(int CertId) throws SQLException
...
-		  softAssert.assertEquals(rs.getString("deviceType"), content.get("deviceType"),"DeviceType");
-		  //softAssert.assertEquals(rs.getString("osUse"),content.get("osUse"),"OsUse");
+		  softAssert.assertEquals(rs.getString("releaseVersion"), content.get("releaseVersion").toString(),"releaseVersion");
+		  softAssert.assertEquals(rs.getString("releaseVersionOld"), content.get("releaseVersionOld").toString(),"releaseVersionOld");
+		  softAssert.assertEquals(rs.getString("osUse"),content.get("osUse"),"OsUse");
 		  softAssert.assertEquals(rs.getString("patch"),content.get("patch"),"patch");
 		  softAssert.assertEquals(rs.getString("firmwareVersion"),content.get("firmwareVersion"),"firmwareVersion");
-		  //softAssert.assertEquals(rs.getString("releaseVersion"),content.get("releaseVersion"),"releaseVersion");
+		  if(driverDetails.isEmpty())
+			  log.info("driver details is empty");
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java]: need to assert that it is empty from DB as well

File: src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java [hunk 12]
```diff
@@ -61,49 +60,101 @@ public void programProductIDAPI(int CertId) throws SQLException
...
+		  softAssert.assertEquals(rs.getString("driverVersion"),driverDetails.get("driverVersion"),"driverVersion");
+		  }
 		  softAssert.assertAll();
             	
+		  
+		    List<String> tierId = jerseyClient.getInnerNodeValuesAsList(sampleJson, "content", "tierDetails", "tierid");
+		    List<String> tier = jerseyClient.getInnerNodeValuesAsList(sampleJson, "content", "tierDetails", "tier");
+		    
+      
+		     List<String> tierIds = jerseyClient.getInnerNodeValuesAsList(sampleJson, "content", "tierDetails", "tierid");
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java]: List<String> tierId is already defined @ line 90

File: src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java [hunk 13]
```diff
@@ -61,49 +60,101 @@ public void programProductIDAPI(int CertId) throws SQLException
...
+          else 
+          {
+            List<String> footnotes = jerseyClient.getInnerNodeValuesAsList(sampleJson, "additionalDetails", "footnotes", "footnote");
+            String[] fNotes = rsF.getString("Footnotes").split("\\|");
+            for(int i = 0; i<footnotes.size(); i++) {
+              log.info("From DB: "+fNotes[i]+" From API: "+footnotes.get(i));
+              softAssert.assertEquals(footnotes.get(i), fNotes[i], "From DB: "+fNotes[i]+" From API: "+footnotes.get(i));
+            }
+          }
+
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java]: kbInfo validation is missing

File: src/test/java/com/vmware/gto/vcg/APItests/VSANIOAPIResponseValidationTests.java [hunk 14]
```diff
@@ -111,21 +111,21 @@ public void programProductIDAPI(int ProductId) throws SQLException {
...
-//          } else {
-//            List<String> footnotes = jerseyClient.getInnerNodeValuesAsList(sampleJson, "additionalDetails", "footnotes", "footnote");
-//            String[] fNotes = rsF.getString("Footnotes").split(",");
-//            for(int i = 0; i<footnotes.size(); i++) {
-//              log.info("From DB: "+fNotes[i]+" From API: "+footnotes.get(i));
-//              softAssert.assertEquals(footnotes.get(i), fNotes[i], "From DB: "+fNotes[i]+" From API: "+footnotes.get(i));
-//            }
-//          }
-//          softAssert.assertAll();
+          String query_FootnotesInformation = vsanioQueries.getFootnotesIformation(Integer.toString(ProductId));
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/VSANIOAPIResponseValidationTests.java]: Need to add releaseVersionOld field validation

File: src/test/java/com/vmware/gto/vcg/APItests/VSANIOCertDetailsValidationTests.java [hunk 15]
```diff
@@ -50,7 +50,7 @@ public void setup() {
 	@Test(dataProvider = "vsanioProducts")
 	public void programProductIDAPI(int CertId) throws SQLException
 	{
-		CertId=4273995;
+		CertId=4282309;
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/VSANIOCertDetailsValidationTests.java]: hardcoded value should be removed

File: src/test/java/com/vmware/gto/vcg/APItests/VSANIOCertDetailsValidationTests.java [hunk 16]
```diff
@@ -60,79 +60,95 @@ public void programProductIDAPI(int CertId) throws SQLException
...
+			          Asserter.assertTrue(featureDetails.get(rs1.getString("featureId")).get("feature").equals(rs1.getString("feature")));
+			          Asserter.assertTrue(featureDetails.get(rs1.getString("featureId")).get("featureValue").equals(rs1.getString("featureValue")));
+			          }
+			      } catch(Exception e) {
+			        log.info(e.getMessage());
+			        Assert.fail();
+			      }
+		  
+            		  
+//		    List<String> featureId = jerseyClient.getInnerNodeValuesAsList(sampleJson, "content", "featureDetails", "featureId");
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/VSANIOCertDetailsValidationTests.java]: remove commented code

File: src/test/java/com/vmware/gto/vcg/dataValidationTests/VSANHDDDataValidationTests.java [hunk 17]
```diff
@@ -1,227 +1,227 @@
...
+				if (!(db_Value == null)) {
+					Assert.assertTrue(ui_getIOModelText.contains(db_Value), "UI doesn not contain " + db_Value);
+				}
+
+			}
+			// Validate UI Release & Feature details and DB Release & Feature
+			// details
+
+			q = vsanhddQueries.getHDDDReleaseDetails(pid);
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/dataValidationTests/VSANHDDDataValidationTests.java]: Why we have this file in this review and the change getHDDDReleaseDetails is incorrect

File: src/test/java/com/vmware/gto/vcg/dbutil/VSANSSDQueries.java [hunk 18]
```diff
@@ -1,288 +1,30 @@
...
+	}
+	
+	
+	
+	
+	
+
+}
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/dbutil/VSANSSDQueries.java]: why other queries are removed

File: src/test/java/com/vmware/gto/vcg/APItests/VSANIOCertDetailsValidationTests.java [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+	            String[] fNotes = rsF.getString("Footnotes").split("\\|");
+	            for(int i = 0; i<footnotes.size(); i++)
+	            {
+	              log.info("From DB: "+fNotes[i]+" From API: "+footnotes.get(i));
+	              softAssert.assertEquals(footnotes.get(i), fNotes[i], "From DB: "+fNotes[i]+" From API: "+footnotes.get(i));
+	            }
+	          }
+		    
+        }
+
```
Comment (ashirvadkum) [inline src/test/java/com/vmware/gto/vcg/APItests/VSANIOCertDetailsValidationTests.java]: Need to add kbInfo validation

File: src/test/java/com/vmware/gto/vcg/APItests/ProgramSearchAPISanityTests.java [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+
+		return stringToDataProvider(vmwProductReleasesAPI);
+	}
+
+	@DataProvider(name = "programNamePartner")
+	public Object[][] serverProducts() {
+		return stringToDataProvider(programNamePartnerAPI);
+	}
+	
+	public Object[][] SQlToDataProvider(String query) {
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/ProgramSearchAPISanityTests.java]: This should be in common functions.

File: src/test/java/com/vmware/gto/vcg/APItests/ProgramSearchAPISanityTests.java [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+			e.printStackTrace();
+		}
+		Object[][] arrayExcelData = new Object[pid.size()][1];
+		for (int i = 0; i < pid.size(); i++) {
+			arrayExcelData[i][0] = pid.get(i);
+		}
+		return arrayExcelData;
+	}
+
+	public Object[][] stringToDataProvider(String inputString) {
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/ProgramSearchAPISanityTests.java]: This should be in common functions.

File: src/test/java/com/vmware/gto/vcg/APItests/ProgramSearchAPISanityTests.java (hunk 20 above)
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/ProgramSearchAPISanityTests.java]: This should be in common functions.

File: src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java [hunk 22]
```diff
@@ -0,0 +1,203 @@
+package com.vmware.gto.vcg.APItests;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+import org.testng.annotations.BeforeClass;
+import org.testng.Assert;
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java]: TestNG assert ?

File: src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java [hunk 23]
```diff
@@ -0,0 +1,203 @@
...
+	}
+
+ 
+	@DataProvider(name = "ssdProducts")
+ public Object[][] ssdProducts() {
+   ResultSet rs = ssdQueries.execQuery("SELECT distinct(c.Component_Id) FROM Components c where Component_Type_Id = 12 Order By Component_Id ASC LIMIT 2;");
+   return SQLResultSetToDataProvider(rs);
+ }
+
+ @DataProvider(name = "pagination")
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDAPIResponseValidationTests.java]: This should be moved to common functions.

File: src/test/java/com/vmware/gto/vcg/APItests/ProductSearchAPISanityTests.java [hunk 24]
```diff
@@ -0,0 +1,249 @@
...
+				+ "LEFT JOIN FTCompatibleSet ft ON cpu.FTComp_Id = ft.FTComp_Id "
+				+ "LEFT JOIN EVCLevels el ON el.EVCLevel_Id = ce.EVCLevel_Id "
+				+ "LEFT JOIN Releases r  ON r.Release_Id  = cr.Release_Id "
+				+ "LEFT JOIN CPU_Identifier ci ON ci.CPUSeries_Id = cpu.CPUSeries_Id "
+				+ "WHERE   (cpu.ShowOnHCL = 1)  AND cpu.ReleaseDate<='2017-10-31' "
+				+ "AND  (r.ShowOnHCL = 1)  AND r.Product_Id in (1,8,9,15) AND r.Release_Id=338 "
+				+ "GROUP BY cpu.CPUSeries_Id ORDER BY cpu.CPUSeries  LIMIT 10;");
+	}
+
+	public Object[][] SQlToDataProvider(String query) {
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/ProductSearchAPISanityTests.java]: Use this method from basePage

File: src/test/java/com/vmware/gto/vcg/APItests/ProductSearchAPISanityTests.java [hunk 25]
```diff
@@ -0,0 +1,249 @@
...
+			e.printStackTrace();
+		}
+		Object[][] arrayExcelData = new Object[pid.size()][1];
+		for (int i = 0; i < pid.size(); i++) {
+			arrayExcelData[i][0] = pid.get(i);
+		}
+		return arrayExcelData;
+	}
+
+	public Object[][] stringToDataProvider(String inputString) {
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/ProductSearchAPISanityTests.java]: Use this method from basePage

File: src/test/java/com/vmware/gto/vcg/APItests/ProgramSearchAPISanityTests.java [hunk 26]
```diff
@@ -0,0 +1,143 @@
...
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.testng.Assert;
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/ProgramSearchAPISanityTests.java]: TestNG Assert ?

File: src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java [hunk 27]
```diff
@@ -0,0 +1,233 @@
...
+		  }
+		  softAssert.assertAll();
+            	
+		  
+		    List<String> tierId = jerseyClient.getInnerNodeValuesAsList(sampleJson, "content", "tierDetails", "tierid");
+		    List<String> tier = jerseyClient.getInnerNodeValuesAsList(sampleJson, "content", "tierDetails", "tier");
+		    
+      
+		     List<String> tierIds = jerseyClient.getInnerNodeValuesAsList(sampleJson, "content", "tierDetails", "tierid");
+		     String[] tIds = rs.getString("tierIds").split("\\|");
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java]: Rename - >dbTierIDs
- > apiTierIds

File: src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java [hunk 28]
```diff
@@ -0,0 +1,233 @@
...
+		    if(tierIds.isEmpty())
+		    {
+		    	System.out.println(rs.getString("tierIds"));
+		    	Asserter.assertTrue(rs.getString("tierIds").equals(""),"tier Ids are not present in API");
+		    }
+		    else
+		    {
+              softAssert.assertEquals(tIds[i],tierId.get(i));
+              softAssert.assertEquals(ts[i],tier.get(i));
+              softAssert.assertAll();
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java]: AssertAll should be at the end.

File: src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java [hunk 29]
```diff
@@ -0,0 +1,233 @@
...
+				          Asserter.assertTrue(featureDetails.get(rs1.getString("featureId")).get("featureValue").equals(rs1.getString("featureValue")));
+				          }
+				      } catch(Exception e) {
+				        log.info(e.getMessage());
+				        Assert.fail();
+				      }
+		     
+         
+		  String query_FootnotesInformation = ssdQueries.certFootnoteDetails(Integer.toString(CertId));
+          ResultSet rsF = ssdQueries.execQuery(query_FootnotesInformation);
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java]: Rename

File: src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java [hunk 30]
```diff
@@ -0,0 +1,233 @@
...
+          rsF.first();
+          if(rsF.getString("Footnotes").equals("\"\"") || rsF.getString("Footnotes").equals(" "))
```
Comment (akamatagi) [inline src/test/java/com/vmware/gto/vcg/APItests/SSDCertDetailsValidationTests.java]: Relook into the code path

**Pull request merged**